app/src/main/java/com/example/sharingang/NewItemFragment.kt

```diff
@@ -28,7 +28,9 @@ class NewItemFragment : Fragment() {
             viewModel.addItem(
                 Item(
                     description = binding.description ?: "",
-                    title = binding.title ?: ""
+                    title = binding.title ?: "",
+                    price = binding.price?.toDoubleOrNull() ?: 0.0
+
                 )

```

Comment:
Same code in both EditItemFragment and NewItemFragment

Reply:
It is indeed the same code, but refactoring this would be quite bothersome, and would maybe lead to unintended results. Both classes use this piece of code, however the `binding` in `NewItemFragment` is different from `binding` in `EditItemFragment`. In one case, it is a `FragmentNewItemBinding` and in the second case it is a `FragmentEditItemBinding`. I was thinking about making a common method that takes a `Binding` as argument and returns the Item based on this argument, but it is quite difficult as Bindings classes are automatically generated and there is no common superclass.